core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -1589,6 +1589,67 @@ private void checkPredicates(RelOptCluster cluster, RelOptTable empTable,
         sortsAs("[IS NULL($0)]"));
   }
 
+  @Test public void testPullUpPredicatesFromUnion() {
+    final RelMetadataQuery mq = RelMetadataQuery.instance();
+    RelNode rel = null;
```

Comment:
Remove the redundant '= null' initializer. Or (my preference) make all variables final and split the method into multiple test methods.

Reply:
Thanks a lot @julianhyde 
I updated and split into several test methods.